### PR TITLE
Remove "passthrough" TLS termination mode

### PR DESCRIPTION
**Pull request by mhutter**
"Passthrough" was originally implemented on the bases that there is no
easy way to
- provide a cert that works with OpenShift Routers in reencrypt modes
- use self-signed certificates with long expiry dates

Hence this commit removes it from the chart, instead making the whole
component MUCH simpler.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by ccremer — CHANGES_REQUESTED** [review]
I think we have misunderstanding of the word "Deprecation".

As outlined in https://syn.tools/commodore/reference/deprecation-policy.html and https://syn.tools/syn/references/review-guide.html a deprecated feature shall still work for a while and only require a minor, non-breaking change. What you did in this PR is completely skip the deprecation, but directly removed support for a feature.

I'm not against the change overall, but please reword the PR and documentation so that there's no mention of deprecation, but rather "removed support/feature" or similar.

File: docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc [hunk 1]
```diff
@@ -6,13 +6,13 @@ This guide describes the steps to perform an upgrade of the component from versi
 
 Configuration of TLS aspect has been overhauled:
 * Separation of internal (ingress to Keycloak) and external (client to ingress) TLS configuration
-
+* Removal of "passthrough" TLS termination mode. "reencrypt" is used instead.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc:9]:
```suggestion
* Removal of `passthrough` TLS termination mode.
  `reencrypt` is used instead.
```

File: docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc [hunk 2]
```diff
@@ -6,13 +6,13 @@ This guide describes the steps to perform an upgrade of the component from versi
...
 
 
 == Step-by-step guide
 
 When upgrading the component, the following changes to the inventory are required:
 
-. Move `tls.termination` to `ingress.tls.termination`
+. Remove `tls.termination`. It is not supported anymore.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc]:
```suggestion
. Remove `tls.termination`.
  It is not supported anymore.
```

File: docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc [hunk 3]
```diff
@@ -6,13 +6,13 @@ This guide describes the steps to perform an upgrade of the component from versi
...
-
+* Removal of "passthrough" TLS termination mode. "reencrypt" is used instead.
 
 
 == Step-by-step guide
 
 When upgrading the component, the following changes to the inventory are required:
 
-. Move `tls.termination` to `ingress.tls.termination`
+. Remove `tls.termination`. It's not supported anymore.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/upgrade-5.x-to-6.x.adoc:16]:
```suggestion
. Remove `tls.termination`.
  It's not supported anymore.
```

**Review by ccremer — APPROVED** [review]
I lost the overview which modes and combinations we now support :see_no_evil: 
I trust that you know what you do :)

Just some styling suggestions (one-line-per-sentence)

**Comment by ccremer** [comment]
@megian IIRC you have this feature activated in one of the cluster. Is this something you can live with?

**Comment by megian** [comment]
> @megian IIRC you have this feature activated in one of the cluster. Is this something you can live with?

Since I had to implement something for the re-encrypt cert (It's a EKS cluster) and don't have the time for. I will try to override it or do a fork of the component for this customer.

**Review by ccremer — CHANGES_REQUESTED** [review]
@megian can you and @mhutter discuss further how to proceed here?
We in Tarazed think that a fork within our own company should be avoided...
Probably we shouldn't merge this PR yet.

**Comment by mhutter** [comment]
@megian My idea was that I migrate the existing passthrough setup (it's a single one) after v6 of this component is released, since it was my understanding that there is no requirement to HAVE a passthrough setup at all

I agree @ccremer that a fork should be avoided at all cost :)

**Comment by megian** [comment]
@mhutter The original idea was to have a end to end encryption to mitigate any man in the middle attacks. Keycloak would be the first I would try to attack. Reencrypt is less secure, as it does open up the potential the ingress controller has some flaws. From a security view - in my opinion -, the highest possible security configuration should be the default of a security component. This is the reason, why I'm not so convinced to prune this option out of the component.

However configuring a manual certificate for the internal cert is a broken concept from my point of view. As it's insecure to have in example a 10 year certificate (especially for a security component like Keycloak) and we currently also don't have any monitoring implemented, for those certificates. 

OpenShift might use https://docs.openshift.com/container-platform/4.9/security/certificate_types_descriptions/service-ca-certificates.html, but my knowledge this is not available on native K8s. https://cert-manager.io/docs/configuration/selfsigned/ might do it.

Manually handling certificates is the worst option. Cryptic and painful to manage. Will check, if there is a better option.

**Comment by mhutter** [comment]
Then I misunderstood the reasons why passthrough was used in the first place, sorry for that.

However since we cannot use certmanager for Passthrough because of keycloak, we can agree that "passthrough" only supports Vault as a certificate source, is that correct?

**Pull request closed without merging**